riscv-page-tables/src/page_table.rs

```diff
@@ -518,6 +518,38 @@ impl<T: PagingMode> PageTableInner<T> {
         entry
     }
 
+    /// Creates a translation for `vaddr` to `paddr` with the given permissions.
+    ///Like map_4k_leaf, but with an added page size parameter.
+    ///
+    /// # Safety
+    ///
+    /// The caller must guarantee that `paddr` references a page uniquely owned by the root
+    /// `GuestStagePageTable`.
+
+    unsafe fn map_general_leaf(
+        &mut self,
+        vaddr: PageAddr<T::MappedAddressSpace>,
+        paddr: SupervisorPageAddr,
+        page_size: PageSize,
+        perms: PteFieldBits,
+    ) -> Result<()> {
+        let entry = self.walk(RawAddr::from(vaddr));
+        use TableEntryType::*;
+        match entry {
+            Locked(l) => {
+                if l.level().leaf_page_size() != page_size {
+                    // maybe not?
+                    return Err(Error::PageSizeNotSupported(l.level().leaf_page_size()));
```

Comment:
A `PageSizeMismatch` is probably a better fit in this case.

Reply:
Fixed